managed/data_source_integration.go

```diff
@@ -0,0 +1,319 @@
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"client_email": {
+						Description: "Client Email",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"client_id": {
+						Description: "Client ID",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"auth_uri": {
+						Description: "Auth URI",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"token_uri": {
+						Description: "Token URI",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"auth_provider_x509_cert_url": {
+						Description: "Auth Provider X509 Cert URL",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"client_x509_cert_url": {
+						Description: "Client X509 Cert URL",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"universe_domain": {
+						Description: "Google Universe Domain",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+				}),
+			},
+		},
+	}, nil
+}
+
+func (r dataSourceIntegrationType) NewDataSource(_ context.Context, p tfsdk.Provider) (tfsdk.DataSource, diag.Diagnostics) {
+	return dataSourceIntegration{
+		p: *(p.(*provider)),
+	}, nil
+}
+
+type dataSourceIntegration struct {
+	p provider
+}
+
+// Read Integration data source
+func (r dataSourceIntegration) Read(ctx context.Context, req tfsdk.ReadDataSourceRequest, resp *tfsdk.ReadDataSourceResponse) {
+
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider wasn't configured before being applied, likely because it depends on an unknown value from another resource.",
+		)
+		return
+	}
+
+	var tpConfig TelemetryProvider
+
+	diags := req.Config.Get(ctx, &tpConfig)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiClient := r.p.client
+
+	accountId, getAccountOK, message := getAccountId(ctx, apiClient)
+	if !getAccountOK {
+		resp.Diagnostics.AddError("Unable to get account ID", message)
+		return
+	}
+
+	projectId, getProjectOK, message := getProjectId(ctx, apiClient, accountId)
+	if !getProjectOK {
+		resp.Diagnostics.AddError("Unable to get the project ID ", message)
+		return
+	}
+
+	telemetryProvider, readOK, message := dataSourceTelemetryProviderRead(accountId, projectId, tpConfig.ConfigName.Value, apiClient)
+	if !readOK {
+		resp.Diagnostics.AddError("Unable to read the state of the integration", message)
+		return
+	}
+
+	diags = resp.State.Set(ctx, &telemetryProvider)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func GetTelemetryProviderByName(accountId string, projectId string, configName string, apiClient *openapiclient.APIClient) (*openapiclient.TelemetryProviderData, error) {
+	resp, _, err := apiClient.TelemetryProviderApi.ListTelemetryProviders(context.Background(), accountId, projectId).Execute()
+
+	if err != nil {
+		return nil, err
```

Comment:
return `GetApiErrorDetails(err)`

Reply:
It is thrown as `GetApiErrorDetails(err)` [here](https://github.com/yugabyte/terraform-provider-ybm/pull/116/files#diff-7507703e65ad40dbead618554dac192c465016d1c6bd3efb341ee93815912bdbR265)